Prac_07/KivyDemos-master/convert_miles_km.py

```diff
@@ -0,0 +1,35 @@
+from kivy.app import App
+from kivy.lang import Builder
+from kivy.core.window import Window
+from kivy.app import StringProperty
+
+MILES_TO_KM = 1.609
+
+
+class ConvertDistanceApp(App):
+    message = StringProperty()
+
+    def build(self):
+        Window.size = (500, 200)
+        self.title = "Converting Distances"
+        self.root = Builder.load_file('convert_miles_km')
+        return self.root
+
+    def convert_miles(self):
+        valid = self.check_valid()
+        self.message = str(valid * MILES_TO_KM)
+
+    def handle_increment(self, change):
+        valid = self.check_valid() + change
+        self.root.ids.user_input.text = str(valid)
+        self.convert_miles()
+
+    def check_valid(self):
```

Comment:
A good function to have (wish i thought of it). For naming, i would consider something along the lines of verify_input or verify_distance so you know the function is checking to see if it is a valid value.